### PR TITLE
[WIP] xandikos: init at 0.0.6

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Motivation for this change

OwnCloud **SUCKS** so much for syncing calendar & contact data, fucks up my data every other day, does not allow creating collections from clients (using vdirsyncer) ... so I need an alternative.

Radicale is **HELL** when trying to set it up,...it just won't work as I request.

So this is another caldav/carddav server which claims to be compatible with my android sync client and vdirsyncer... hopefully it is.

---

Still waiting to get this issue: https://github.com/jelmer/xandikos/issues/70 resolved, though

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

WIP!

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
So it looks like `dulwich` gets build with `fastimport`, which is _not_ python3 in nixpkgs, but `xandikos` needs/is python3.

Can someone tell me what to do?

**Comment by matthiasbeyer** [comment]
Issue solved. This works now, but I would like to add a service definition for this before it gets merged!

**Comment by matthiasbeyer** [comment]
Added a service definition, though I don't know how to test this properly or even implement VM tests for it.

Can someone jump in? :smile:

File: nixos/modules/services/networking/xandikos.nix
Comment (FRidh) [inline nixos/modules/services/networking/xandikos.nix]: Au

File: nixos/modules/services/networking/xandikos.nix
Comment (FRidh) [inline nixos/modules/services/networking/xandikos.nix]: dot

File: nixos/modules/services/networking/xandikos.nix
Comment (FRidh) [inline nixos/modules/services/networking/xandikos.nix]: No need for parentheses.

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by matthiasbeyer** [comment]
Done. Still no tests, though.

**Comment by matthiasbeyer** [comment]
Can someone suggest how to test this?

**Comment by Infinisil** [comment]
https://github.com/pimutils/vdirsyncer seems like an easy way to test this.

**Comment by ivan** [comment]
For the author, reviewers, and committers: this PR was scanned and appears to add a use of the deprecated `types.string`, which emits a warning as of #66346. Before merging, please change this to another type, possibly:

- `types.str` for a single string where merging does not make sense, or cannot work
- `types.lines` for multi-line configuration or scripts where merging is possible
- `types.listOf types.str` for a mergeable list of strings

File: nixos/modules/misc/ids.nix
Comment (0x4A6F) [inline nixos/modules/misc/ids.nix]: Can we utilize systemd's `DynamicUser` here?

**Pull request closed without merging**
